app/Services/UserService.php

```diff
@@ -2,24 +2,59 @@
 
 namespace App\Services;
 
+use Firebase\JWT\{JWT, Key};
+use App\Models\{User, Invite, UserRole};
 use Illuminate\Support\Facades\Auth;
 use Illuminate\Validation\UnauthorizedException;
 
 class UserService
 {
-    public function authenticateUser(array $validatedUserLogin): string
+    public function authenticateUser(array $validatedUserLogin): array
     {
         if (!Auth::attempt($validatedUserLogin)) {
             throw new UnauthorizedException("Invalid login credentials");
         }
 
         $user = Auth::user();
 
-        return $user->createToken('authToken')->plainTextToken;
+        $token = $user->createToken('authToken')->plainTextToken;
+
+        return [
+            'user' => $user,
+            'token' => $token
+        ];
     }
 
     public function revokeUserToken(): void
     {
         Auth::user()->currentAccessToken()->delete();
     }
+
+    public function storeUser(array $validatedStoreUser): User
+    {
+        $decodedToken = JWT::decode(
+            $validatedStoreUser['token'],
+            new Key(config('jwt.secret_key'), config('jwt.algorithm'))
+        );
+
+        $decodedTokenArray = (array) $decodedToken;
+
+        $invitedUser = Invite::where(
+            [
+                ['email', $decodedTokenArray['email']],
+                ['name', $decodedTokenArray['name']]
+            ]
+        )->first();
+
+        $createdUser = User::create($validatedStoreUser);
+
+        UserRole::create([
```

Comment:
@sabinshrestha8 use the relationship to fill the role

Reply:
https://github.com/proshore/ProshoreAssetManagement-Backend/pull/12/commits/3066181296383d9a27aad9a27cddb5c238e13b38